data-prepper-plugins/key-value-processor/src/test/java/com/amazon/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java

```diff
@@ -0,0 +1,237 @@
+        assertThat(editedRecords.get(0), notNullValue());
+        assertThat(editedRecords.get(0).getData().toJsonString(),
+                equalTo("{\"message\":\"key1  =value1\",\"parsed_message\":{\"key1\":\"value1\"}}"));
+    }
+
+    @Test
+    void testTrimValueRegexKvProcessor() {
+        //override setup
+        keyValueProcessor = new KeyValueProcessor(pluginMetrics, getConfig("", null, "&", "=", "", "\\s"));
+
+        final Map<String, Object> testData = new HashMap();
+        testData.put("message", "key1=value1   &key2=value2");
+        final Record<Event> record = buildRecordWithEvent(testData);
+
+        final List<Record<Event>> editedRecords = (List<Record<Event>>)keyValueProcessor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.size(), equalTo(1));
+        assertThat(editedRecords.get(0), notNullValue());
+        assertThat(editedRecords.get(0).getData().toJsonString(),
+                equalTo("{\"message\":\"key1=value1   &key2=value2\",\"parsed_message\":{\"key1\":\"value1\",\"key2\":\"value2\"}}"));
+    }
+
+    @Test
+    void testTrimValueAndKeyRegexKvProcessor() {
+        //override setup
+        keyValueProcessor = new KeyValueProcessor(pluginMetrics, getConfig("", null, "&", "=", "\\s", "\\s"));
+
+        final Map<String, Object> testData = new HashMap();
+        testData.put("message", "key1  =value1   & key2 = value2 ");
+        final Record<Event> record = buildRecordWithEvent(testData);
+
+        final List<Record<Event>> editedRecords = (List<Record<Event>>)keyValueProcessor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.size(), equalTo(1));
+        assertThat(editedRecords.get(0), notNullValue());
+        assertThat(editedRecords.get(0).getData().toJsonString(),
+                equalTo("{\"message\":\"key1  =value1   & key2 = value2 \",\"parsed_message\":{\"key1\":\"value1\",\"key2\":\"value2\"}}"));
+    }
+
+    @Test
+    void testShutdownIsReady() {
+        assertThat(keyValueProcessor.isReadyForShutdown(), is(true));
+    }
+
+    private void reflectivelySetField(final KeyValueProcessorConfig keyValueProcessorConfig, final String fieldName, final Object value) throws NoSuchFieldException, IllegalAccessException {
+        final Field field = KeyValueProcessorConfig.class.getDeclaredField(fieldName);
+        try {
+            field.setAccessible(true);
+            field.set(keyValueProcessorConfig, value);
+        } finally {
+            field.setAccessible(false);
+        }
+    }
+
+    private KeyValueProcessorConfig getConfig(final String prefix, final String nonMatchValue,
+                                              final String fieldDelimiterRegex, final String keyValueDelimiterRegex, final String trimKeyRegex,
+                                              final String trimValueRegex) {
+        KeyValueProcessorConfig config = new KeyValueProcessorConfig();
```

Comment:
This would be better with a mock object. Then you don't need to use reflection (Mockito will).